docs/source/formulations_example.rst

```diff
@@ -0,0 +1,648 @@
+The template includes a comprehensive list of all allowed names and labels.
+
+.. code-block:: python
+
+    from citrine.resources.process_template import ProcessTemplate
+
+    mix_template = dataset.process_templates.register(
+        ProcessTemplate(
+            "mix"
+            allowed_names=["simple syrup", "sugar", "water", "lime juice", "triple sec", "tequila", "ice"],
+            allowed_labels=["sweetener", "acid", "alcohol", "simple syrup"]
+        )
+    )
+
+Notice that "simple syrup" is an allowed ingredient name, but we do not distinguish between "simple syrup A" and "simple syrup B."
+This reflects the fact that we only ever use one simple syrup in a mixing process.
+The resulting table will have one column indicating which simple syrup was used and a second column indicating how much was used.
+Distinguishing in this way between the unique id of the material ("simple syrup A," "simple syrup B," etc.) and the generic name of the ingredient ("simple syrup") is especially useful when there are many materials to choose from.
+
+To fill out the example, we illustrate some of the objects involved in specifying the spec for a particular margarita recipe.
+This assumes that the material specs for the raw ingredients and the simple syrups have already been uploaded.
+
+.. code-block:: python
+
+    from citrine.resources.process_spec import ProcessSpec
+    from citrine.resources.ingredient_spec import IngredientSpec
+    from citrine.resources.material_spec import MaterialSpec
+    from gemd.entity.value import NominalReal
+
+    mix_margarita_spec = dataset.process_specs.register(
+        ProcessSpec(f"mix margarita B", template=mix_template)
+    )
+    dataset.ingredient_specs.register(
+        IngredientSpec(
+            "simple syrup",
+            material=simple_syrup_B_spec,
+            process=mix_margarita_spec,
+            labels=["simple syrup"]
+            mass_fraction=NominalReal(nominal=0.15, units="")
+        )
+    )
+    dataset.ingredient_specs.register(
+        IngredientSpec(
+            "triple sec",
+            material=triple_sec_spec,
+            process=mix_margarita_spec,
+            labels=["alcohol", "sweetener"],
+            mass_fraction=NominalReal(nominal=0.1, units="")
+        )
+    )
+    # register remaining ingredient specs....
+    # Then register the resulting material spec.
+    margarita_spec = dataset.material_specs.register(
+        MaterialSpec(f"margarita B", process=mix_margarita_spec, template=margarita_template)
+    )
+
+This material spec is then fed as the sole ingredient into a "blend margarita B" process spec, which produces a "blended margarita B" material spec.
+A measurement spec is attached to the material spec to measure "tastiness."
+Finally, run objects are created corresponding to each spec, to represent what actually happened.
```

Comment:
Can we get an example?

Reply:
I'd rather not. Since ingesting GEMD is not the point of this example, I only want to include the minimum information that is necessary to make the rest of the tutorial make sense.